mesonbuild/dependencies/base.py

```diff
@@ -505,51 +505,61 @@ def __init__(self, name, environment, kwargs, language=None):
         else:
             for_machine = MachineChoice.HOST
 
-        # Create a nested function for sake of early return
+        # Create an iterator of options
         def search():
-            # Only search for the pkg-config for each machine the first time and
-            # store the result in the class definition
-            if PkgConfigDependency.class_pkgbin[for_machine] is None:
-                mlog.debug('Pkg-config binary for %s is not cached.' % for_machine)
-            else:
-                mlog.debug('Pkg-config binary for %s is cached.' % for_machine)
-                choice = PkgConfigDependency.class_pkgbin[for_machine]
-                assert choice is not None
-                return choice
             # Lookup in cross or machine file.
-            bt = environment.binaries[for_machine]
-            potential_pkgpath = bt.lookup_entry('pkgconfig')
-            if potential_pkgpath is None:
-                mlog.debug('Pkg-config binary missing from cross or native file, or PKG_CONFIG undefined.')
-            else:
-                mlog.debug('Pkg-config binary for %s specified from config file as %s.', for_machine, potential_pkgpath)
-                potential_pkgbin = ExternalProgram.from_entry('pkgconfig', potential_pkgpath)
-                if not potential_pkgbin.found():
-                    mlog.debug(
-                        'Pkg-config %s for machine %s specified at %s but not found.',
-                        potential_pkgbin.name, for_machine, potential_pkgbin.command)
-                else:
-                    return potential_pkgbin
+            potential_pkgpath = environment.binaries[for_machine] \
+                .lookup_entry('pkgconfig')
```

Comment:
It's odd that this is pulled down onto a new line, we don't seem to have any preference for short lines in meson, and the next line is very long anyway.